### PR TITLE
[BUG 🐞] DOMA-2176 Fixed contact deletedAt scenarios

### PR DESCRIPTION
**Pull request by SavelevMatthew**
**Problem**: Cannot create same contact, if it was deleted in organization
**Solution**: Fix bad-written validation logic

### CODE REVIEW

File: apps/condo/domains/contact/schema/Contact.js [hunk 1]
```diff
@@ -96,13 +96,14 @@ const Contact = new GQLListSchema('Contact', {
...
-            const { property, unitName, name, phone, deletedAt } = resolvedData
+            const newItem = { ...existingItem, ...resolvedData }
+            const { property, unitName, name, phone } = newItem
             const [contact] = await ContactAPI.getAll(context, {
                 property: { id: property },
                 unitName,
                 name,
                 phone,
-                deletedAt,
+                deletedAt: null,
```
Comment (pahaz) [inline apps/condo/domains/contact/schema/Contact.js:106]: It’s by default here. You can just remove this key

Comment (SavelevMatthew) [reply apps/condo/domains/contact/schema/Contact.js:106]: Changed to native keystone request, where deletedAt is required

Comment (pahaz) [reply apps/condo/domains/contact/schema/Contact.js:106]: It looks like a ADR4 violdtion. You add the `deletedAt` inside the business logic.

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/contact/schema/Contact.js [hunk 2]
```diff
@@ -96,13 +96,14 @@ const Contact = new GQLListSchema('Contact', {
     },
     hooks: {
         validateInput: async ({ resolvedData, operation, existingItem, addValidationError, context }) => {
-            const { property, unitName, name, phone, deletedAt } = resolvedData
+            const newItem = { ...existingItem, ...resolvedData }
+            const { property, unitName, name, phone } = newItem
             const [contact] = await ContactAPI.getAll(context, {
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/contact/schema/Contact.js]: moved to `find`

**Pull request merged**